### PR TITLE
Handled pre-round equivocation bug

### PR DESCRIPTION
**Pull request by gavraz**

### CODE REVIEW

File: hare/preroundtracker.go [hunk 1]
```diff
@@ -29,20 +29,24 @@ func (pre *PreRoundTracker) OnPreRound(msg *pb.HareMessage) {
 		return
 	}
 
-	// only handle first pre-round msg
-	if _, exist := pre.preRound[verifier.String()]; exist {
+	var sToTrack *Set = nil
```
Comment (antonlerner) [inline hare/preroundtracker.go]: you can refactor the else using better arrangement of parameters, sToTrack can start with msg.Message.Values as initial value and be modified else

Comment (gavraz) [reply hare/preroundtracker.go]: I think it is a matter of style but do understand it is a more simple and go-idiomatic approach so I changed it to fit the no else and use if for non-happy flows

**Review by antonlerner — APPROVED** [review]

File: hare/preroundtracker.go [hunk 2]
```diff
@@ -7,14 +7,14 @@ import (
 )
 
 type PreRoundTracker struct {
-	preRound  map[string]struct{} // maps PubKey->Pre-Round msg (unique)
-	tracker   *RefCountTracker    // keeps track of seen values
-	threshold uint32              // the threshold to prove a single value
+	preRound  map[string]*Set  // maps PubKey->Pre-Round msg (unique)
```
Comment (zalmen) [inline hare/preroundtracker.go]: I don't understand the comment - what is unique? and should it be Set instead of "Pre-Round msg"?

Comment (zalmen) [reply hare/preroundtracker.go]: Also (general comment) - please add tests to test these changes. It is suspicious that you a major logical change in the PreroundTracker without needing to change any of the test code, please make sure that this struct is well tested.

**Review by zalmen — CHANGES_REQUESTED** [review]

**Review by zalmen — APPROVED** [review]

**Pull request merged**